assembly/src/procedures/mod.rs

```diff
@@ -176,6 +178,32 @@ impl AsRef<str> for ProcedureName {
     }
 }
 
+impl Serializable for ProcedureName {
+    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
+        let name_bytes = self.name.as_bytes();
+        let num_bytes = name_bytes.len() as u8;
+        if num_bytes > MAX_PROC_NAME_LEN {
+            return Err(SerializationError::LengthTooLong);
+        } else {
+            target.write_u8(num_bytes);
+            target.write_bytes(name_bytes);
+        }
```

Comment:
nit: this else is not really needed here since the `if.true` ends with a block termination.

for better readability:

```rust
        }
        target.write_u8(num_bytes);
        target.write_bytes(name_bytes);
```

Reply:
Fixed!